web/src/components/pages/contribution/speak/speak.tsx

```diff
@@ -475,17 +494,61 @@ class SpeakPage extends React.Component<Props, State> {
 
   private agreeToTerms = async () => {
     this.setState({ showPrivacyModal: false });
+    this.setState({ privacyAgreedChecked: true });
     this.props.updateUser({ privacyAgreed: true });
     this.upload(true);
   };
 
+  private onPrivacyAgreedChange = (privacyAgreed: boolean) => {
+    this.setState({ privacyAgreedChecked: privacyAgreed });
+  };
+
   private toggleDiscardModal = () => {
     this.setState({
       showPrivacyModal: false,
       showDiscardModal: !this.state.showDiscardModal,
     });
   };
 
+  private handleSubmit = (evt: React.SyntheticEvent) => {
+    const { user } = this.props;
+
+    const hasSeenFirstCTA = window.sessionStorage.getItem(SEEN_FIRST_CTA);
+    const hasSeenSecondCTA = window.sessionStorage.getItem(SEEN_SECOND_CTA);
+
+    this.props.updateUser({ privacyAgreed: this.state.privacyAgreedChecked });
+
+    evt.preventDefault();
+    this.upload(this.state.privacyAgreedChecked);
+
+    // display first CTA screen if it has not been seen it before
+    // and the user does not have an account
+    if (hasSeenFirstCTA !== 'true' && !user.account) {
+      this.setState({ shouldShowFirstCTA: true });
+      window.sessionStorage.setItem(SEEN_FIRST_CTA, 'true');
+    }
+
+    // display second CTA screen if it has not been seen it before
+    // and the user does not have an account
+    if (
+      hasSeenFirstCTA === 'true' &&
+      hasSeenSecondCTA !== 'true' &&
+      !user.account
+    ) {
+      this.setState({ shouldShowSecondCTA: true });
+      window.sessionStorage.setItem(SEEN_SECOND_CTA, 'true');
+    }
+
+    // Reset for unauthenticated users who have seen the first and second CTA so they can see new clips to record
+    if (
+      !user.account &&
```

Comment:
since `!user.account` is a condition thats being checked 3 times in the same function, it might make it easier to read to break up the chain of if statments like: 
```javascript
if (!user.account){
  if(otherCondition){}
  else if (otherCondition){}
//restartState etc
}```